### ISSUE TITLE
Simply campaign doesn't send email

### ISSUE THREAD

**Issue by andreab79** [mautic/mautic#5289]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Yes
| Feature request? | 
| Enhancement? | 

## Description:
Campaign doesn't send email to segment contacts. Cron jobs are correctly scheduled

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.11.0
| PHP version | 5.6 or 7.0 (tested both)

### Steps to reproduce:
1. Create a campaign to send email (immediately, after time or in a specific date, tested both)
2. Campaign doesn't send email to segment contacts
 
### Log errors: 

no log content in main folder and in app folder

**Comment by escopecz** [comment]
There must be some misconfiguration. It works for everybody else. Please make sure you have some contacts in the source segment who have an email address. Does email sending works normally? Like if you send test email or segment email? Do you use queue to send emails? What email provider do you use? Does it work with another provider? What is the output of command `app/console mautic:campaigns:trigger`?

**Comment by andreab79** [comment]
hi, tx for your reply. really works for you??? are you sure you are speaking of 2.11.0 version? please look https://www.mautic.org/community/index.php/8718-2-10-1-2-11-0-emails-not-sending

anyway as i said cron jobs are configured correctly following this video https://www.youtube.com/watch?v=XXMLk8TrG10

**Comment by escopecz** [comment]
Ok, I tested it and when I execute `app/console mautic:campaigns:update` to add the contact from segment to the campaign and `app/console mautic:campaigns:trigger` to execute the campaign actions. Then I receive the emails.

Cron jobs configuration  may be different for every server. Please to through all of my questions and find an answer. This article can help you with that: https://mautic.org/docs/en/tips/troubleshooting.html

**Comment by andreab79** [comment]
tx for your time. sure cron scripts are different for server and folder path. i used my configuration and adding a email address for notifications i receive email messages related to tasks done correctly. but if i check on mautic, for example segments are empty (no contacts). also if i add manually contacts to segments and launch campaign, doesn't send email

**Comment by andreab79** [comment]
i followed your advices

1. deleted cache
2. updated schema
3. checked log in system info and shows these errors

[2017-11-06 03:00:15] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused (uncaught exception) at /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:social:monitoring` [] []
[2017-11-06 03:00:16] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused (uncaught exception) at /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:messages:send` [] []
[2017-11-06 03:00:17] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2017-11-06 11:00:19] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:social:monitoring` [] []
[2017-11-06 11:00:19] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:messages:send` [] []
[2017-11-06 11:00:19] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []

**Comment by escopecz** [comment]
Try to modify 'db_host' from "localhost" to "127.0.0.1" in app/config/local.php and clear the cache.

**Comment by andreab79** [comment]
done, wait 1 hour for cron jobs but still not works

**Comment by escopecz** [comment]
I'm afraid you'll have to clone the campaign because the current campaign events will be marked as failed and they won't trigger again.

**Comment by andreab79** [comment]
done, for campaigns and segments. not works, i mean segments are not populated and doesn't send email

**Comment by escopecz** [comment]
Segments are not populated? Do you run the `app/console mautic:segments:update` command? There is definitely something wrong with your cron jobs. Consider installing [Cronfig](https://cronfig.io) so you could clearly see what's going on with your background jobs.

**Comment by andreab79** [comment]
cronfig folder uploaded but in plugin page, when i click cronfig icon shows me a popup with "Takes care of the cron jobs - makes your Mautic alive." and not email field. so doesn't send me api :(

**Comment by escopecz** [comment]
Try to follow the documentation more closely. https://docs.cronfig.io/integrations/mautic

**Comment by andreab79** [comment]
yes i followed exactly this. please can we consider in private way a quote ti fix my installation?

**Comment by escopecz** [comment]
No, you didn't. Based on the output you wrote you see you clicked on the plugin icon. Not on the admin menu item. There is even screenshot in the docs on what to click.

**Comment by raribeiro** [comment]
Same error here, is not sending any email after upgrading to 2.11.0 .. (campaigns and segments). The SES connection test works perfectly in the settings.

**Comment by Rikk03** [comment]
I have the same problem SES connection, test emails work fine, triggers campaign send then I find error icon on individuals details page.  Nothing shown in error logs or anywhere else.  Just nothing received.

**Comment by escopecz** [comment]
Isn't the problem in SES then? Can you test with another provider to confirm?

**Comment by Rikk03** [comment]
Triggering events for campaign 4
Triggering first level events
4 total events(s) to be processed in batches of 100
 0/4 [>---------------------------]   0%
 1/4 [=======>--------------------]  25%
 2/4 [==============>-------------]  50%
 3/4 [=====================>------]  75%
 4/4 [============================] 100%
4 event(s) executed

Triggering scheduled events
0 total events(s) to be processed in batches of 100
0 event(s) executed

Triggering 'non-action' events
4 total contact(s) to be analyzed in batches of 100
 0/8 [>---------------------------]   0%
 8/8 [============================] 100%
0 event(s) executed

**Comment by Rikk03** [comment]
@andreab79  are you also using SES?

**Comment by raribeiro** [comment]
With **Sparkpost running normally**, the problem is with SES. I just tested.

**Comment by Rikk03** [comment]
Ok well question here is whether anything was changed in the last update that would effect the send to SES?

**Comment by robwent** [comment]
I'm not seeing any issues with ses on several installs.

I'm using the ses option in the email dropdown configuration and not 'other smtp' settings.

**Comment by robwent** [comment]
Ignore that, I'm still on 2.10.1

**Comment by Rikk03** [comment]
Ok well thats a good start .... what changed between 2.10.1 and the latest version which has mucked up the SES send.

**Comment by andreab79** [comment]
hi, tx to all for contribution. i don't use ses, just a vps on hostgator.

finally i used correctly cronfig, escopecz has right, i didn't :), 3 cron for free profile: Update segments, Update campaigns and Trigger campaigns but segments are not updated and emails campaign are not sent. i share log error of cronfig, alwais the same error for both jobs

HTTP status is not 2xx. That means not successful.
Job output
error: secret key is missing in the request
Job request headers
{
	"date": "Tue, 07 Nov 2017 11:23:02 GMT",
	"server": "Apache",
	"cache-control": "no-cache",
	"set-cookie": [
		"xxx=xxx; path=/; HttpOnly"
	],
	"keep-alive": "timeout=5, max=100",
	"connection": "Keep-Alive",
	"transfer-encoding": "chunked",
	"content-type": "text/plain; charset=UTF-8"
}

**Comment by Rikk03** [comment]
OK well since we found a problem elsewhere I guess it should remain about SES, sorry for hijacking your post, however please add more details next time!

**Comment by escopecz** [comment]
@andreab79 try to log out of cronfig, then Mautic and log back in to both. Delete the jobs you created and create them again. It's a cronfig issue.

**Comment by andreab79** [comment]
sorry i have to delete also cron created in my vps?

**Comment by andreab79** [comment]
hi now with cronfig it works!!! any idea to use cron jobs of my vps instead cronfig? 
here have scripts configured in my vps. apparently they work because if i set my email address i receive correctly notification updates

0,15,30,45	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:segments:update	 Edit      Delete
5,20,35,50	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:campaigns:rebuild
10,25,40,55	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:campaigns:trigger
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:emails:send
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:messages:send
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:social:monitoring
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:email:fetch
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:webhooks:process
0	0	1	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:iplookup:download

**Comment by escopecz** [comment]
If you are 100% sure they work then you don't need cronfig. I suggested it just so you could easily see the output.

**Comment by andreab79** [comment]
you mean output in log? just this
[07-Nov-2017 16:00:20 Europe/Rome] PHP Warning:  PDO::query(): Error reading result set's header in /home/webagile/public_html/MARKETINGAGILE/webagile/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 104

**Comment by lvnilesh** [comment]
Hi @andreab79 Can you please paste output of your `crontab -l` in a gist so I can read clearly? 

    ```
    Please paste crontab -l   output here in between these pair of tripple backticks.
    ```

**Comment by andreab79** [comment]
`
0,15,30,45	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:segments:update
5,20,35,50	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:campaigns:rebuild
10,25,40,55	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:campaigns:trigger
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:emails:send
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:messages:send
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:social:monitoring
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:emailfetch
0,30	*	*	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:webhooks:process
0	0	1	*	*	/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:iplookup:download
`

**Comment by robwent** [comment]
If you are using cpanel then you can add an email address to the cron area and it will email you the results each time the cron runs so you can check the emails for error messages

**Comment by andreab79** [comment]
robwent you have right, i did and when cron run i receive correctly email notification like this 
```
Ricostruire i contatti per il segmento 1
0 contatti totale da aggiungere in gruppi di 300
 contatti totale da rimuovere in gruppi di 300
0 contatti interessati

Ricostruire i contatti per il segmento 11
```
but in mautic nothing happens

**Comment by robwent** [comment]
Doesn't that notification say that there is nothing to do? If so, there
wouldn't be any changes in Mautic admin.

**Comment by lvnilesh** [comment]
This is not going to resolve your issue but just check this.

```
app/console mautic:emailfetch
```
should be
```
app/console mautic:email:fetch
```

**Comment by lvnilesh** [comment]
Also, try running this command on the shell and paste me the output
```
/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:emails:send
```

**Comment by andreab79** [comment]
sorry Ivnilesh, i have cpanel or whm, i don't have skills to use shell

**Comment by andreab79** [comment]
error notifications

/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:social:monitoring
```
[Doctrine\DBAL\Exception\ConnectionException]
  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused



  [Doctrine\DBAL\Driver\PDOException]
  SQLSTATE[HY000] [2002] Connection refused



  [PDOException]
  SQLSTATE[HY000] [2002] Connection refused


mautic:social:monitoring [--mid [MID]] [--batch-size BATCH-SIZE] [--query-count [QUERY-COUNT]] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command>
```

/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:segments:update
```
 [Doctrine\DBAL\Exception\ConnectionException]
  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused



  [Doctrine\DBAL\Driver\PDOException]
  SQLSTATE[HY000] [2002] Connection refused



  [PDOException]
  SQLSTATE[HY000] [2002] Connection refused


mautic:segments:update [-b|--batch-limit [BATCH-LIMIT]] [-m|--max-contacts [MAX-CONTACTS]] [-i|--list-id [LIST-ID]] [-f|--force] [-t|--timeout TIMEOUT] [-x|--lock_mode LOCK_MODE] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command>
```

/opt/cpanel/ea-php56/root/usr/bin/php /home/webagile/public_html/MARKETINGAGILE/webagile/app/console mautic:messages:send
```
Processamento della coda messaggi


  [Doctrine\DBAL\Exception\ConnectionException]
  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused



  [Doctrine\DBAL\Driver\PDOException]
  SQLSTATE[HY000] [2002] Connection refused



  [PDOException]
  SQLSTATE[HY000] [2002] Connection refused


mautic:messages:send [-c|--channel [CHANNEL]] [-i|--channel-id CHANNEL-ID] [-m|--message-id MESSAGE-ID] [-f|--force] [-t|--timeout TIMEOUT] [-x|--lock_mode LOCK_MODE] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command>
```

**Comment by andreab79** [comment]
but i don't this this is the problem because from 2 hour no more error notifications. other ideas?

**Comment by lvnilesh** [comment]
Database `connection refused` means your mautic install is unhealthy. You are better off fixing that first before spending time with other ideas.

**Comment by andreab79** [comment]
hi, as i said before, from hours not anymore error log, mautic works correctly except for cron. other ideas?

**Comment by Rikk03** [comment]
Update, sending email via Queue does not work.  Otherwise it works.

**Comment by andreab79** [comment]
in my case not work both, immediate and queque. i'm thinking if it is possible convert standard cron job in curl script, i'd like test this. could you help me to convert?

**Comment by Rikk03** [comment]
this is two completely different cases, yours is a configuration problem NOT a but issue that this post is now about.  Im only talking about connection and send with Amazon SES

**Comment by andreab79** [comment]
ahahaha sorry but who changed post title??? originally we were not speaking about ses.

**Comment by Rikk03** [comment]
This forum is not for configuration issues, I expect thats why he changed
it.

**Comment by andreab79** [comment]
you decided by yourself this is a configuration issue? wow, please tell also how many child i will have :). please don't reply for sport, i'm testing all aspects to define which is the problem, i'm not alone to have problem with 2.11 version

**Comment by andreab79** [comment]
after 3 days i solved simply updating php version to 7.0. i think would be better in mautic requirements write: 'PHP 7.0 recommended' :)

tx to all for support

**Comment by Rikk03** [comment]
OK well back to SES ........ perhaps its time to switch to SES API @escopecz ? What of this contribution https://github.com/mautic/mautic/pull/5263

**Comment by andreab79** [comment]
hi @escopecz , sorry but the same error again after update to version 2.12.1
cron jobs work correctly, for example segments are updated but doesn't send campaign emails
also email configuration is ok because sends single email to contact

tx for support!

**Comment by nheljhoncap** [comment]
@andreab79 Did you fixed the issue on v.2.12.1  ? I have also the same problem

**Comment by andreab79** [comment]
@nheljhoncap i solved simply updating php version to 7.0.
